libtransact/src/state/merkle.rs

```diff
@@ -47,11 +48,11 @@ type StateHash = Vec<u8>;
 
 #[derive(Clone)]
 pub struct MerkleState {
-    db: LmdbDatabase,
+    db: Arc<dyn Database>,
```

Comment:
If `clone`, as described above, is implemented, this Arc implementation isn't needed.